spec/lib/saml/post_url_service_spec.rb

```diff
@@ -613,8 +613,27 @@
             let(:expected_log_message) { 'Redirecting to /terms-of-use' }
             let(:expected_log_payload) { { type: :ssoe } }
 
+            shared_examples 'client list' do |env, list|
+              before { allow(Settings).to receive(:vsp_environment).and_return(env) }
+
+              it 'returns the appropriate list of clients depending on the environment' do
+                result = subject.send(:enabled_tou_clients)
+                expect(result)
+                  .to eq(list)
```

Comment:
It's not really best practice to test private methods directly. You should structure your tests in a way so the private method is 'tested' by calling a public method. Plus, the thing we really want to test here is if we get the expected `redirect_url`